dash-renderer/src/components/core/DocumentTitle.react.js

```diff
@@ -7,16 +7,13 @@ class DocumentTitle extends Component {
         super(props);
         const {update_title} = props.config;
         this.state = {
-            initialTitle: document.title,
             update_title,
         };
     }
 
     UNSAFE_componentWillReceiveProps(props) {
         if (this.state.update_title && props.isLoading) {
             document.title = this.state.update_title;
-        } else {
```

Comment:
While this allows the clientside callback to set the value, it seems to break existing tests / behavior. Would a better verification be to online revert back to `inititalTitle` if the value is `update_title` instead?

Reply:
Thanks! I totally misinterpreted what `update_title` meant. I believe https://github.com/plotly/dash/pull/1343/commits/e190cc941b89275677f0b3b995d21d5220e0209b is the logic we're looking for

Reply:
I think it'd be safer to bail out entirely (before even the `props.isLoading` test) if `!update_title`. Then this entire component would be a noop, as it should be in that case; callbacks can do whatever they want with `document.title` and we won't interfere.

When you _do_ have an `update_title`, I may agree with @Marc-Andre-Rivet that there needs to be a test when we're reverting back to the non-updating title. Consider a callback chain, where the first callback changes `document.title`, but we're still in a loading state so `DocumentTitle` doesn't receive new props and we don't pick up the changed title into `this.state`. Then after the second callback we reset `document.title` to `this.state.title`, losing the callback-provided title.

Seems to me the logic you have right now is fine during loading, but when reverting to `this.state.title` below it would be better to do something like:
```js
if (document.title === this.state.update_title) {
    document.title = this.state.title;
} else {
    this.setState({title: document.title});
}
```

Reply:
Thanks @alexcjohnson @Marc-Andre-Rivet ! I believe I have addressed your comments and added the appropriate tests now.